### PR TITLE
csi: storageclass update to gp2-csi

### LINKED ISSUE #14535: storageClassName: gp2 should be rename to storageClassName: gp2-csi
**Is this a bug report or feature request?**
* Bug Report

For cluster-on-pvc.yaml example the storageclass is set to gp2, but AWS have moved to gp2-csi and gp3-csi, so kind change the names in the examples and document

**Deviation from expected behavior:**

**Expected behavior:**

**How to reproduce it (minimal and precise):**

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary

**Logs to submit**:

* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

  To get logs, use `kubectl -n <namespace> logs <pod name>`
  When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
  Read [GitHub documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Cluster Status to submit**:

* Output of kubectl commands, if necessary

  To get the health of the cluster, use `kubectl rook-ceph health`
  To get the status of the cluster, use `kubectl rook-ceph ceph status`
  For more details, see the [Rook kubectl Plugin](https://rook.io/docs/rook/latest-release/Troubleshooting/kubectl-plugin)

**Environment**:
* OS (e.g. from /etc/os-release):
* Kernel (e.g. `uname -a`):
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod):
* Storage backend version (e.g. for ceph do `ceph -v`):
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest-release/Troubleshooting/ceph-toolbox/#interactive-toolbox)):

### PR DESCRIPTION
**Pull request by ShravaniVangur**
Updating storageClassName for cluster-on-pvc.yaml file from gp2 to gp2-csi.

Resolves #14535

**Cluster Status:**
For rook-ceph health: 
image

For rook-ceph cluster status: 
image

Pods: 
image

### CODE REVIEW

File: deploy/examples/cluster-on-pvc.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ spec:
     # size appropriate for monitor data will be used.
     volumeClaimTemplate:
       spec:
-        storageClassName: gp2
+        storageClassName: gp2-csi
         resources:
           requests:
```
Comment (parth-gr) [inline deploy/examples/cluster-on-pvc.yaml:33]: update this comment 
`        # Currently, "gp2" has been identified as such`

File: deploy/examples/cluster-on-pvc.yaml [hunk 2]
```diff
@@ -132,7 +132,7 @@ spec:
                 requests:
                   storage: 10Gi
               # IMPORTANT: Change the storage class depending on your environment
-              storageClassName: gp2
+              storageClassName: gp2-csi
               volumeMode: Block
               accessModes:
                 - ReadWriteOnce
```
Comment (parth-gr) [inline deploy/examples/cluster-on-pvc.yaml:138]: Update the cluster-stretched-aws.yaml

plus there is a pvc-cluster.md docuemtnation file,
If you do a search of `gp2` you will find couple of instances, so update wherever they are related

**Review by parth-gr — CHANGES_REQUESTED** [review]

File: deploy/examples/cluster-on-pvc.yaml (hunk 2 above)
Comment (ShravaniVangur) [reply deploy/examples/cluster-on-pvc.yaml:138]: I have updated the pvc-cluster.md documentation file in places which were related with the previous changes. Please do let me know if any other changes need to be made in this file.

Comment (parth-gr) [reply deploy/examples/cluster-on-pvc.yaml:138]: You can also change
create_test.go -> integration tests && root-test-framework.md
ceph-cluster-crd.md ->doc

**Review by travisn — COMMENTED** [review]
LGTM, please just squash the commits.

**Review by parth-gr — APPROVED** [review]
lgtm

**Review by travisn — APPROVED** [review]

**Pull request merged**